### PR TITLE
Jenkinsyaml

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
@spxtr I am making a few changes to this PR based on my experiments with pushing them to jenkins.  I will update and add a PTAL along with links to the jobs pushed from the configs.

**Comment by pwittrock** [comment]
PTAL

I generate the xml locally and pushed to jenkins.  The 5 jobs this yaml creates are visible under the view:

http://kubekins.dls.corp.google.com/view/Node/

File: hack/jenkins/job-configs/global.yaml [hunk 1]
```diff
@@ -85,6 +85,33 @@
...
 
+- builder:
+    name: docker-build-publish
+    builders:
+        - raw:
+            xml: |
+               <com.cloudbees.dockerpublish.DockerBuilder plugin="docker-build-publish@1.1">
+                   <server plugin="docker-commons@1.2"/>
+                   <registry plugin="docker-commons@1.2">
+                       <credentialsId>---</credentialsId>
```
Comment (spxtr) [inline hack/jenkins/job-configs/global.yaml]: For this I would probably put it in a variable with a comment describing how to find it on Jenkins.

Comment (pwittrock) [inline hack/jenkins/job-configs/global.yaml]: How would the variable get populated with the value we want when the jjb sync is run?

Comment (spxtr) [inline hack/jenkins/job-configs/global.yaml]: I would just hard-code in the value to a variable in the YAML. There are plenty of places where our config can't be ported exactly to other Jenkins instances, so I wouldn't stress this too much.

File: hack/jenkins/job-configs/global.yaml [hunk 2]
```diff
@@ -85,6 +85,33 @@
...
 
+- builder:
+    name: docker-build-publish
+    builders:
+        - raw:
+            xml: |
+               <com.cloudbees.dockerpublish.DockerBuilder plugin="docker-build-publish@1.1">
+                   <server plugin="docker-commons@1.2"/>
+                   <registry plugin="docker-commons@1.2">
+                       <credentialsId>1f361efb-5b85-4f61-91a7-4ec7fb2a5c23</credentialsId>
```
Comment (spxtr) [inline hack/jenkins/job-configs/global.yaml]: I'd appreciate a comment above this section explaining what this is.

**Comment by spxtr** [comment]
LGTM otherwise.

**Comment by spxtr** [comment]
LGTM. I'll manually merge after lunch.

File: hack/jenkins/job-configs/global.yaml (hunk 2 above)
Comment (pwittrock) [inline hack/jenkins/job-configs/global.yaml]: Done

File: hack/jenkins/job-configs/global.yaml (hunk 1 above)
Comment (pwittrock) [inline hack/jenkins/job-configs/global.yaml]: Done

**Comment by pwittrock** [comment]
SGTM.  Thanks!

**Comment by spxtr** [comment]
@k8s-oncall going to manually merge this, it's not touched by CI and shouldn't break anything critical.

**Pull request merged**